sharding/indexHashedNodesCoordinator_test.go

```diff
@@ -500,6 +502,125 @@ func TestIndexHashedGroupSelector_ComputeValidatorsGroupTest6From10ValidatorsSho
+		list = append(list, mock.NewValidatorMock(big.NewInt(0), 0, []byte("pk"+strconv.Itoa(i)), []byte("addr"+strconv.Itoa(i))))
+	}
+
+	nodesMap := make(map[uint32][]sharding.Validator)
+	nodesMap[0] = list
+
+	getCounter := 0
+	putCounter := 0
+
+	cache := &mock.NodesCoordinatorCacheMock{
+		PutCalled: func(key []byte, value interface{}) (evicted bool) {
+			putCounter++
+			return false
+		},
+		GetCalled: func(key []byte) (value interface{}, ok bool) {
+			getCounter++
+			return nil, false
+		},
+	}
+
+	arguments := sharding.ArgNodesCoordinator{
+		ShardConsensusGroupSize: consensusGroupSize,
+		MetaConsensusGroupSize:  1,
+		Hasher:                  &mock.HasherMock{},
+		NbShards:                1,
+		Nodes:                   nodesMap,
+		SelfPublicKey:           []byte("key"),
+		ConsensusGroupCache:     cache,
+	}
+
+	ihgs, _ := sharding.NewIndexHashedNodesCoordinator(arguments)
+
+	miniBlocks := 10
+
+	for i := 0; i < miniBlocks; i++ {
+		for j := 0; j <= i; j++ {
+			randomness := strconv.Itoa(j)
+			list2, _ := ihgs.ComputeValidatorsGroup([]byte(randomness), uint64(j), 0)
+
+			assert.Equal(t, consensusGroupSize, len(list2))
+		}
+	}
+
+	computationNr := miniBlocks * (miniBlocks + 1) / 2
+
+	assert.Equal(t, computationNr, getCounter)
+	assert.Equal(t, computationNr, putCounter)
+}
+
+func TestIndexHashedGroupSelector_ComputeValidatorsGroup400of400For10BlocksMemoization(t *testing.T) {
+	consensusGroupSize := 400
+	list := make([]sharding.Validator, 0)
+
+	//generate 400 validators
+	for i := 0; i < 400; i++ {
+		list = append(list, mock.NewValidatorMock(big.NewInt(0), 0, []byte("pk"+strconv.Itoa(i)), []byte("addr"+strconv.Itoa(i))))
+	}
+
+	nodesMap := make(map[uint32][]sharding.Validator)
+	nodesMap[0] = list
+
+	getCounter := 0
+	putCounter := 0
+
+	//consensusGroup := list[0:21]
+	cacheMap := make(map[string]interface{})
+	cache := &mock.NodesCoordinatorCacheMock{
+		PutCalled: func(key []byte, value interface{}) (evicted bool) {
+			putCounter++
+			strKey := string(key)
+			cacheMap[strKey] = value
+			return false
+		},
+		GetCalled: func(key []byte) (value interface{}, ok bool) {
+			getCounter++
+			strKey := string(key)
+			val, ok := cacheMap[strKey]
+			if ok {
+				return val, true
+			}
+			return nil, false
+		},
+	}
+
+	arguments := sharding.ArgNodesCoordinator{
+		ShardConsensusGroupSize: consensusGroupSize,
+		MetaConsensusGroupSize:  1,
+		Hasher:                  &mock.HasherMock{},
+		NbShards:                1,
+		Nodes:                   nodesMap,
+		SelfPublicKey:           []byte("key"),
+		ConsensusGroupCache:     cache,
+	}
+
+	ihgs, _ := sharding.NewIndexHashedNodesCoordinator(arguments)
+
+	miniBlocks := 10
+
+	for i := 0; i < miniBlocks; i++ {
+		for j := 0; j <= i; j++ {
+			randomness := strconv.Itoa(j)
+			list2, _ := ihgs.ComputeValidatorsGroup([]byte(randomness), uint64(j), 0)
+
+			assert.Equal(t, consensusGroupSize, len(list2))
+		}
+	}
+
+	computationNr := miniBlocks * (miniBlocks + 1) / 2
```

Comment:
shouldn't this be a lower value? It is exactly the same as in  `TestIndexHashedGroupSelector_ComputeValidatorsGroup400of400For10locksNoMemoization`

Reply:
The computationNr is the number of gets. It should be the same, but the number of puts is just miniBlocks, 10. The actual computation is done 10 times.